integration-scripts/install_cppstats.sh

```diff
@@ -1,23 +1,39 @@
-#!/bin/sh
+#!/bin/bash
+# NOTE: bash instead of sh is required for pushd/popd
 
 export CPPSTATS_VERSION=0.8.4
 
 echo "Providing cppstats $CPPSTATS_VERSION"
 
-mkdir -p vendor/
-cd vendor/
+pushd .
+TMPDIR=`mktemp -d` || exit 1
+cd ${TMPDIR}
 
-wget --quiet https://codeload.github.com/clhunsen/cppstats/tar.gz/v$CPPSTATS_VERSION -O /tmp/cppstats.tar.gz
-tar -xvf /tmp/cppstats.tar.gz
+CPPSTATS_URL="https://codeload.github.com/clhunsen/cppstats/tar.gz/v${CPPSTATS_VERSION}"
+wget --quiet ${CPPSTATS_URL} -O ${TMPDIR}/cppstats.tar.gz
+if [ ! -e ${TMPDIR}/cppstats.tar.gz ]
+then
+    echo "Could not download cppstats from ${CPPSTATS_URL}"
+    exit 1
+fi
+tar -xvf cppstats.tar.gz
 export CPPSTATS=$PWD/cppstats-$CPPSTATS_VERSION/
 echo '#!/bin/bash' > $CPPSTATS/cppstats
 echo "cd $CPPSTATS" >> $CPPSTATS/cppstats
 echo "PYTHONPATH=\"\$PYTHONPATH:$CPPSTATS/lib\" ./cppstats.py \"\$@\"" >> $CPPSTATS/cppstats
 chmod +x $CPPSTATS/cppstats
-wget --quiet http://131.123.42.38/lmcrs/beta/srcML-Ubuntu12.04-64.tar.gz -O /tmp/srcML.tar.gz
-tar -xvf /tmp/srcML.tar.gz
+
+echo "Providing srcML"
+SRCML_URL="http://131.123.42.38/lmcrs/beta/srcML-Ubuntu12.04-64.tar.gz"
```

Comment:
Am 16/06/2016 um 23:47 schrieb Claus Hunsen:
> In integration-scripts/install_cppstats.sh
> <https://github.com/siemens/codeface/pull/50#discussion_r67430345>:
> 
>>  export CPPSTATS=$PWD/cppstats-$CPPSTATS_VERSION/
>>  echo '#!/bin/bash' > $CPPSTATS/cppstats
>>  echo "cd $CPPSTATS" >> $CPPSTATS/cppstats
>>  echo "PYTHONPATH=\"\$PYTHONPATH:$CPPSTATS/lib\" ./cppstats.py \"\$@\"" >> $CPPSTATS/cppstats
>>  chmod +x $CPPSTATS/cppstats
>> -wget --quiet http://131.123.42.38/lmcrs/beta/srcML-Ubuntu12.04-64.tar.gz -O /tmp/srcML.tar.gz
>> -tar -xvf /tmp/srcML.tar.gz
>> +
>> +echo "Providing srcML"
>> +SRCML_URL="http://131.123.42.38/lmcrs/beta/srcML-Ubuntu12.04-64.tar.gz"
> 
> As said in 08e047b
> <https://github.com/siemens/codeface/commit/08e047b488458da70d4451692d6b0a0ea0229445>,
> cppstats is not compatible with the current beta of srcML. The download
> location of choice is
> |http://131.123.42.38/lmcrs/srcML-Ubuntu12.04-64.tar.gz|.

thanks for the information, I'll update the patch.

Thanks, Wolfgang
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/siemens/codeface/pull/50/files/08e047b488458da70d4451692d6b0a0ea0229445..a2d81beffaacc4069b8b45ffcdd118ad1cf616f0#r67430345>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABbE1__1RNZ6v_q1uY1xGq_4Irspa2qTks5qMcSBgaJpZM4I3yGy>.
>